controllers/global_search_setup.go

```diff
@@ -0,0 +1,554 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	searchv1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	searchGlobal       = "search-global"
+	searchGlobalConfig = "search-global-config"
+)
+
+var (
+	multiclusterGlobalHubGvr = schema.GroupVersionResource{
+		Group:    "operator.open-cluster-management.io",
+		Version:  "v1alpha4",
+		Resource: "multiclusterglobalhubs",
+	}
+	multiclusterengineResourceGvr = schema.GroupVersionResource{
+		Group:    "multicluster.openshift.io",
+		Version:  "v1",
+		Resource: "multiclusterengines",
+	}
+	managedClusterResourceGvr = schema.GroupVersionResource{
+		Group:    "cluster.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "managedclusters",
+	}
+	managedServiceAccountGvr = schema.GroupVersionResource{
+		Group:    "authentication.open-cluster-management.io",
+		Version:  "v1beta1",
+		Resource: "managedserviceaccounts",
+	}
+	manifestWorkGvr = schema.GroupVersionResource{
+		Group:    "work.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "manifestworks",
+	}
+)
+
+// Reconcile Global Search.
+//  1. Check the global-search-preview annotation.
+//  2. Validate pre-requisites.
+//     a. MulticlusterGlobalHub operator is installed in the cluster.
+//     b. The ManagedServiceAccount add-on is enabled in the MultiClusterEngine CR.
+//     c. The ClusterProxy addon is enabled in the MultiClusterEngine CR.
+//  3. Enable global search feature in the console.
+//  4. Enable federated search feature in the search-api deployment.
+//  5. Create a ManagedServiceAccount recource for each Managed Hub.
+//  6. Create a ManifestWork resource for each Managed Hub.
+func (r *SearchReconciler) reconcileGlobalSearch(ctx context.Context,
+	instance *searchv1alpha1.Search) (*reconcile.Result, error) {
+
+	if instance.ObjectMeta.Annotations["search.open-cluster-management.io/global-search-peview"] == "true" ||
```

Comment:
```suggestion
	if instance.ObjectMeta.Annotations["search.open-cluster-management.io/global-search-preview"] == "true" ||
```

Reply:
ahhh, this explains a lot of dev errors.

I added the fully qualified annotation for consistency, but I like `global-search-preview` because it's simpler. Thoughts?

Reply:
Shorter and lesser chance of errors. :)

Reply:
Will we remove the preview part for GA?

Reply:
My plan for GA is to move it to the spec.